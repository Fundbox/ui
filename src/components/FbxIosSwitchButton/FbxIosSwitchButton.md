# FbxIosSwitchButton

## Basic usage

- Pass any HTML attribute or native DOM event to `fbx-ios-switch-button`
- By default, @change will be debounced by 1500ms. To disable that, just pass :debounceDisabled

```html
<fbx-ios-switch-button
    name="my-button"
    id="my-id"
    data-qa="my-ios-switch-button-data-qa"
    tabindex="3"
    @change="onChange"
>
</fbx-ios-switch-button>
```



