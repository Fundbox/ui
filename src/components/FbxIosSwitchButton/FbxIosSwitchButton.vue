<template>
  <div class="container">
    <label class="switch">
      <input
        class="input"
        type="checkbox"
        v-bind="$attrs"
        v-on="listeners"
        :value="value"
        :checked="value"
      >
      <span class="slider round"></span>
    </label>
  </div>
</template>

<script>
import debounce from 'lodash/debounce'

export default {
  name: 'FbxIosSwitchButton',
  props: ['value', 'debounceDisabled'],
  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: (event) => {
          this.debounceDisabled ? this.$emit('input', event.target.checked) : debounce(() => {
            this.$emit('input', event.target.checked)
          }, 1500)
        }
      }
    }
  },
}
</script>

<style lang="scss" scoped>
  @import "../../styles/utils/color-palette";
  @import "../../styles/utils/mixins.scss";
  .switch {
    margin: 0;
    position: relative;
    display: inline-block;
    width: 39px;
    height: 22px;
  }

  .input {
    display: none;
    &:checked + .slider {
      background-color: $dark-green;
    }

    &:focus + .slider {
      box-shadow: 0 0 1px $dark-green;
    }

    &:checked + .slider:before {
      transform: translateX(16px);
    }

  }

  .slider {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: $medium-gray;
    transition: $fast-ease;
    &:before {
      position: absolute;
      content: "";
      height: 16px;
      width: 16px;
      left: 4px;
      bottom: 3px;
      background-color: $white;
      transition: $fast-ease;
    }
    &.round {
      border-radius: 34px;
      &:before {
        border-radius: 50%;
      }
    }
  }
</style>
