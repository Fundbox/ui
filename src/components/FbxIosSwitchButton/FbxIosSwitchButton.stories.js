import { storiesOf } from '@storybook/vue'
import { withInfo } from 'storybook-addon-vue-info'

import FbxIosSwitchButton from './FbxIosSwitchButton.vue'
import summary from './FbxIosSwitchButton.md'
// import {action} from "@storybook/addon-actions/src/index";

const stories = storiesOf('Components/IosSwitchButton', module)

stories.add('default', withInfo({ summary })(() => ({
  components: { FbxIosSwitchButton },
  data() { return { value: true, debounceDisabled: true } },
  watch: {
    value(val) {
      console.log(`New value: ${val}`);
    },
  },
  template: `<fbx-ios-switch-button debounceDisabled="debounceDisabled" v-model="value"></fbx-ios-switch-button>`
})))
