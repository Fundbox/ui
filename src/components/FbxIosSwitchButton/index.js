import FbxIosSwitchButton from './FbxIosSwitchButton.vue'
import { registerComponents, vueUse } from '../../utils/plugins'

const components = {
  FbxIosSwitchButton
}

const VuePlugin = {
  install(Vue) {
    registerComponents(Vue, components)
  }
}

vueUse(VuePlugin)

export default VuePlugin
