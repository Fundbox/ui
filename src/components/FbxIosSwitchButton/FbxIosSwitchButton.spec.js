import { shallowMount } from '@vue/test-utils'
import FbxIosSwitchButton from './FbxIosSwitchButton.vue'

describe('Components/FbxIosSwitchButton', () => {
  describe('snapshots', () => {
    it('renders the default correctly', () => {
      const wrapper = shallowMount(FbxIosSwitchButton, {
        // mounting options https://vue-test-utils.vuejs.org/api/options.html
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
