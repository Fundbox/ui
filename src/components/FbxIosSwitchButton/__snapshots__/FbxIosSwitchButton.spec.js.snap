// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/FbxIosSwitchButton snapshots renders the default correctly 1`] = `<div class="container"><label class="switch"><input type="checkbox" class="input" value="false"> <span class="slider round"></span></label></div>`;
